src/test/scala/za/co/absa/abris/avro/read/confluent/SchemaManagerSpec.scala

```diff
@@ -27,18 +28,88 @@ class SchemaManagerSpec extends FlatSpec with BeforeAndAfter {
+    )
+    assert(subject + "-value" == SchemaManager.getSubjectName(subject, false, (null, null), conf))
+    assert(subject + "-key" == SchemaManager.getSubjectName(subject, true, (null, null), conf))
+  }
+
+  it should "retrieve the correct subject name for RecordName strategy" in {
+    val subject = "a_subject"
+    val conf = Map(
+      SchemaManager.PARAM_VALUE_SCHEMA_NAMING_STRATEGY -> SchemaManager.SchemaStorageNamingStrategies.RECORD_NAME,
+      SchemaManager.PARAM_KEY_SCHEMA_NAMING_STRATEGY -> SchemaManager.SchemaStorageNamingStrategies.RECORD_NAME
+    )
+
+    val schemaName = "schema_name"
+    val schemaNamespace = "schema_namespace"
+
+    assert(s"$schemaNamespace.$schemaName" == SchemaManager.getSubjectName(subject, false, (schemaName, schemaNamespace), conf))
+    assert(s"$schemaNamespace.$schemaName" == SchemaManager.getSubjectName(subject, true, (schemaName, schemaNamespace), conf))
+  }
+
+  it should "retrieve the null for RecordName strategy if schema is null" in {
     val subject = "a_subject"
-    assert(subject + "-value" == SchemaManager.getSubjectName(subject, false))
-    assert(subject + "-key" == SchemaManager.getSubjectName(subject, true))
+    val conf = Map(
+      SchemaManager.PARAM_VALUE_SCHEMA_NAMING_STRATEGY -> SchemaManager.SchemaStorageNamingStrategies.RECORD_NAME,
+      SchemaManager.PARAM_KEY_SCHEMA_NAMING_STRATEGY -> SchemaManager.SchemaStorageNamingStrategies.RECORD_NAME
+    )
+
+    val schemaName = null
+    val schemaNamespace = "namespace"
+
+    assert(null == SchemaManager.getSubjectName(subject, false, (schemaName, schemaNamespace), conf))
+    assert(null == SchemaManager.getSubjectName(subject, true, (schemaName, schemaNamespace), conf))
+  }
+
+  it should "retrieve the correct subject name for TopicRecordName strategy" in {
+    val topic = "a_subject"
+    val conf = Map(
+      SchemaManager.PARAM_VALUE_SCHEMA_NAMING_STRATEGY -> SchemaManager.SchemaStorageNamingStrategies.TOPIC_RECORD_NAME,
+      SchemaManager.PARAM_KEY_SCHEMA_NAMING_STRATEGY -> SchemaManager.SchemaStorageNamingStrategies.TOPIC_RECORD_NAME
+    )
+
+    val schemaName = "schema_name"
+    val schemaNamespace = "schema_namespace"
+
+    assert(s"$topic-$schemaNamespace.$schemaName" == SchemaManager.getSubjectName(topic, false, (schemaName, schemaNamespace), conf))
+    assert(s"$topic-$schemaNamespace.$schemaName" == SchemaManager.getSubjectName(topic, true, (schemaName, schemaNamespace), conf))
   }
 
   it should "not try to configure Schema Registry client if parameters are empty" in {
     SchemaManager.configureSchemaRegistry(Map[String,String]())
-    assertResult(false) {SchemaManager.isSchemaRegistryConfigured()} // should still be unconfigured
+    assertResult(false) {SchemaManager.isSchemaRegistryConfigured} // should still be unconfigured
+  }
+
+  it should "retrieve the null for TopicRecordName strategy if schema is null" in {
```

Comment:
I would move this test case up by one so it's next to it's `TopicRecordName` companion test case. Right now there's an unrelated test case keeping them apart.